go/parquet/schema/reflection.go

```diff
@@ -0,0 +1,827 @@
+			key := strings.TrimSpace(strings.ToLower(kv[0]))
+			value := strings.TrimSpace(kv[1])
+
+			switch key {
+			case "name":
+				info.Name = value
+			case "type":
+				info.Type = typeFromStr(value)
+			case "keytype":
+				info.KeyType = typeFromStr(value)
+			case "valuetype":
+				info.ValueType = typeFromStr(value)
+			case "length":
+				info.Length = int32FromType(value)
+			case "keylength":
+				info.KeyLength = int32FromType(value)
+			case "valuelength":
+				info.ValueLength = int32FromType(value)
+			case "scale":
+				info.Scale = int32FromType(value)
+			case "keyscale":
+				info.KeyScale = int32FromType(value)
+			case "valuescale":
+				info.ValueScale = int32FromType(value)
+			case "precision":
+				info.Precision = int32FromType(value)
+			case "keyprecision":
+				info.KeyPrecision = int32FromType(value)
+			case "valueprecision":
+				info.ValuePrecision = int32FromType(value)
+			case "fieldid":
+				info.FieldID = int32FromType(value)
+			case "keyfieldid":
+				info.KeyFieldID = int32FromType(value)
+			case "valuefieldid":
+				info.ValueFieldID = int32FromType(value)
+			case "repetition":
+				info.RepetitionType = repFromStr(value)
+			case "valuerepetition":
+				info.ValueRepetition = repFromStr(value)
+			case "converted":
+				info.Converted = convertedFromStr(value)
+			case "keyconverted":
+				info.KeyConverted = convertedFromStr(value)
+			case "valueconverted":
+				info.ValueConverted = convertedFromStr(value)
+			case "logical":
+				info.LogicalFields["type"] = value
+			case "keylogical":
+				info.KeyLogicalFields["type"] = value
+			case "valuelogical":
+				info.ValueLogicalFields["type"] = value
+			default:
+				switch {
+				case strings.HasPrefix(key, "logical."):
+					info.LogicalFields[strings.TrimPrefix(key, "logical.")] = value
+				case strings.HasPrefix(key, "keylogical."):
+					info.KeyLogicalFields[strings.TrimPrefix(key, "keylogical.")] = value
+				case strings.HasPrefix(key, "valuelogical."):
+					info.ValueLogicalFields[strings.TrimPrefix(key, "valuelogical.")] = value
+				}
+			}
+		}
+		info.UpdateLogicalTypes()
+		return &info
+	}
+	return nil
+}
+
+// typeToNode recurseively converts a physical type and the tag info into parquet Nodes
+//
+// to avoid having to propagate errors up potentially high numbers of recursive calls
+// we use panics and then recover in the public function NewSchemaFromStruct so that a
+// failure very far down the stack quickly unwinds.
+func typeToNode(name string, typ reflect.Type, repType parquet.Repetition, info *taggedInfo) Node {
+	// set up our default values for everything
+	var (
+		converted             = ConvertedTypes.None
+		logical   LogicalType = NoLogicalType{}
+		fieldID               = int32(-1)
+		physical              = parquet.Types.Undefined
+		typeLen               = 0
+		precision             = 0
+		scale                 = 0
+	)
+	if info != nil { // we have struct tag info to process
+		fieldID = info.FieldID
```

Comment:
just a note there has been a recent issue/PR that is changing the logic around fieldID (c++ code will not generate them any more)

Reply:
Should I modify this to no longer auto-generate the fieldIDs? is there a specific reason why the C++ code won't generate them anymore?

Reply:
See PR #10289 for details, essentially the  field IDs in thrift are mean for other system to set for schema evolution/conversion purposes.

Reply:
Yes.  No specific reason outside of reducing complexity/noise.  Arrow simply does not have enough information to generate anything meaningful here (the current "field order" assignment is not really informative and could be easily regenerated by a user if they truly needed it).

Instead the C++ implementation will just pass through the value to/from the parquet layer.

Reply:
Fair enough. I'm fine with pulling out the automatic field id generation here. The current implementation i have here would persist user set ids and only auto generates if they are left as -1, but it's easy to just leave the -1's or pass through what is given. I'll comment after I push that change.